src/ripple/shamap/README.md

```diff
@@ -1,22 +1,33 @@
 # SHAMap Introduction #
 
-July 2014
+March 2020
 
 The SHAMap is a Merkle tree (http://en.wikipedia.org/wiki/Merkle_tree).
 The SHAMap is also a radix tree of radix 16
 (http://en.wikipedia.org/wiki/Radix_tree).
 
-*We need some kind of sensible summary of the SHAMap here.*
+A SHAMap is a tree with two node types:
```

Comment:
```suggestion
A SHAMap is a trie with two node types:
```

I'm not sure which nomenclature is better, but as far as I know a SHAmap has more characteristics of a trie than a tree.

Reply:
I think it should be trie 👍

Reply:
I'll make this substitution throughout the doc.

Reply:
> I'm not sure which nomenclature is better, but as far as I know a SHAmap has more characteristics of a trie than a tree.

@carlhua @MarkusTeufelberger @HowardHinnant 
A trie is a type of tree. I think tree is actually more appropriate and correct here, because a SHAMap is a tree that has properties of a trie as well as a merkle tree. If we say a SHAMap is a trie, we are ignoring the merkle tree aspect of the SHAMap. I vote for tree here, and throughout the document, unless we are specifically talking about the aspects of the SHAMap that are a trie.

Reply:
Which property of a merkle tree are you referring to?

Reply:
The hashing. Each node's hash is derived from the hashes of its children.